src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpClient,
+  HttpErrorResponse,
+  HttpHeaders,
+} from '@angular/common/http';
+import { catchError, Observable, throwError } from 'rxjs';
+import { Router } from '@angular/router';
+import { ErrorService } from 'src/app/services/error.service';
+import { EditUserData, User } from 'src/app/models/models';
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  /* private apiUrl = 'http://localhost:3000'; */
+  private apiUrl = 'https://final-task-backend-production-4453.up.railway.app';
+
+  private tokenKey: string | null = null;
+  public username: string | null = null;
+  public userId: string | null = null;
+  jwtHelperService = new JwtHelperService();
+
+  constructor(
+    private http: HttpClient,
+    private router: Router,
+    private errorService: ErrorService
+  ) {}
+
+  signUp(formData: User): Observable<any> {
+    return this.http
+      .post<any>(`${this.apiUrl}/auth/signup`, formData)
+      .pipe(catchError(this.errorHandler.bind(this)));
+  }
+
+  signIn(formData: User): Observable<any> {
+    return this.http
+      .post<any>(`${this.apiUrl}/auth/signin`, formData)
+      .pipe(catchError(this.errorHandler.bind(this)));
+  }
+
+  saveToken(login: string, token: string) {
+    localStorage.setItem('login', login);
```

Comment:
I would recommend moving logic related to localStorage to separate service. In case, the project migrate to some other storage you cat fix all necessary staff in one place